test/aleph/http/client_test.clj

```diff
@@ -0,0 +1,43 @@
+(ns aleph.http.client-test
+  (:require [aleph.http.client :as client]
+            [clojure.test :as t :refer [deftest is testing]]))
+
+(deftest test-domain-extracting
+  (testing "ASCII domain is extracted from URL as is."
+    (let [uri (client/req->domain {:url "https://domainname.com:80"})]
+      (is (= "https" (.getScheme uri)))
+      (is (= "domainname.com" (.getHost uri)))
+      (is (= 80 (.getPort uri)))))
+  (testing "ASCII domain is extracted from request host as is."
+    (let [uri (client/req->domain {:scheme "https"
+                                   :host "domainname.com"
+                                   :port 80})]
+      (is (= "https" (.getScheme uri)))
+      (is (= "domainname.com" (.getHost uri)))
+      (is (= 80 (.getPort uri)))))
+  (testing "ASCII domain is extracted from request server name as is."
+    (let [uri (client/req->domain {:scheme "https"
+                                   :server-name "domainname.com"
+                                   :port 80})]
+      (is (= "https" (.getScheme uri)))
+      (is (= "domainname.com" (.getHost uri)))
+      (is (= 80 (.getPort uri)))))
+  (testing "IDN domain in URL is translated to ASCII."
+    (let [uri (client/req->domain {:url "https://名がドメイン.com:80"})]
+      (is (= "https" (.getScheme uri)))
+      (is (= "xn--v8jxj3d1dzdz08w.com" (.getHost uri)))
+      (is (= 80 (.getPort uri)))))
+  (testing "IDN domain from request host is translated to ASCII."
```

Comment:
Could be good to add a test for an [Internationalized TLD](https://en.wikipedia.org/wiki/Internationalized_country_code_top-level_domain) too.